packages/clerk-js/src/ui/utils/phoneUtils.ts

```diff
@@ -115,3 +116,26 @@ export function stringToFormattedPhoneString(str: string): string {
   const parsed = parsePhoneString(str);
   return `+${parsed.code} ${formatPhoneNumber(parsed.number, parsed.pattern, parsed.code)}`;
 }
+
+type CountryData = {
+  searchTerm: string;
+  country: CountryEntry;
+};
+
+export function getValidCountry(value: string, options: CountryData[]) {
```

Comment:
🙃 It's worth adding a comment for this best-effort algorithm to find the longest country code. Alternatively, you can rename this function to getLongestValidCountryCode to make it more descriptive.